frb_dart/lib/src/env/io.dart

```diff
@@ -0,0 +1,33 @@
+// see https://github.com/flutter/flutter/issues/55870
+
+import 'dart:io';
+
+const truthyValues = ['true', '1', 'yes'];
+
+/// ✅ correctly read e.g. export JSON=true && ...
+/// ❌ fail at reading e.g. dart --define=SILICON=true ...
+/// ❌ fail at reading e.g. dart -DSILICON=true ...
+bool? readEnvFromPlatform(String envVarName) =>
+    (Platform.environment[envVarName] != null &&
+            Platform.environment[envVarName]!.isNotEmpty)
+        ? truthyValues.contains(Platform.environment[envVarName]!.toLowerCase())
+        : null;
+
+/// ✅ correctly read e.g. dart --define=SILICON=true ...
+/// ✅ correctly read e.g. dart -DSILICON=true ...
+/// ❌ fail at reading e.g. export JSON=true && ...
+bool? readEnvFromPrimitive(String envVarName) =>
+    bool.hasEnvironment(envVarName) ? bool.fromEnvironment(envVarName) : null;
+
+bool readBoolEnv(String envVarName, {bool defaultValue = false}) =>
+    readEnvFromPrimitive(envVarName) ??
+    readEnvFromPlatform(envVarName) ??
+    defaultValue;
+
+final isWeb = readBoolEnv('dart.library.html');
+final int? portEnv = Platform.environment['PORT'] != null
+    ? int.parse(Platform.environment['PORT']!)
+    : null;
+final isSilicon = readBoolEnv('SILICON');
+final useJSON = readBoolEnv('JSON');
```

Comment:
Oops maybe we should not put "useJson" in frb_dart, which is for everyone, not for benchmark

Reply:
ah I forgot... I think this is because only on `web` with dart2js I have to feed the env vars as cli args only to the server `serve.dart`, since WASM has no access to any env var. I'll check if there's another way.

Reply:
That is reasonable, just do not be in frb_dart but in frb_example/benches

Take your time!

Reply:
so it's fine to copy/paste `serve.dart` in benches example ?

Reply:
Oh that is a problem, surely not copy-paste. See my comments below

Reply:
below ?

Reply:
Weird, I have replied it somewhere indeed... (maybe above?)

The idea is, (1) not copy-paste (2) can we change the code below, not `if (useJSON) '--dart-define=JSON=true'`, but let serve.dart completely not knowing the json flag, and let the *user* pass in `--dart-define` by himself via the `Opts.rest(args)`

![image](https://user-images.githubusercontent.com/5236035/194731216-5204501c-22cd-4abf-954b-07987e4e65d2.png)

Reply:
That's a fair solution indeed!